### PR TITLE
DLS-12535:[CGT] Improve error handling of duplicate filenames for supporting evidence

### PR DESCRIPTION
**Pull request by ZaraFarzan**

### CODE REVIEW

**Review by vamsikrishnapujari — DISMISSED** [review]

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceController.scala [hunk 1]
```diff
@@ -325,35 +326,49 @@ class SupportingEvidenceController @Inject() (
...
-              upscanUpload =>
-                upscanUpload.upscanCallBack match {
-                  case Some(_: UpscanSuccess) =>
-                    Redirect(
-                      routes.SupportingEvidenceController.checkYourAnswers()
+              {
+                case (upscanUpload, DuplicateUpscanFileName) =>
+                  BadRequest(
+                    uploadPage(
+                      upscanUpload,
```
Comment (mahesh-hmrc) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceController.scala]: This should not use the previous success reference, but initiate a new upscan request and then use that reference instead. At the moment we risk using the stale/completed upscan metadata for the next one if we don't re-initiate it.

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/supportingevidence/upload.scala.html [hunk 2]
```diff
@@ -76,14 +78,35 @@ <h2 class="govuk-heading-m">@messages(s"$key.heading2")</h2>
...
 }
 
+@duplicateFileNameErrorMessage = @{
+    if(hasDuplicateFileNameError)
+        Some(messages(s"$key.duplicateFileName"))
+    else
+        None
+}
+
 @layout(title, backLinkUrl = Some(backLink.url)) {
```
Comment (mahesh-hmrc) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/supportingevidence/upload.scala.html]: we need to pass `hasDuplicateFileNameError` to the `hasErrors` property here. Otherwise page title won't be marked as error.

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala [hunk 3]
```diff
@@ -1248,7 +1248,7 @@ class SupportingEvidenceControllerSpec
 
           val upscanSuccess        = sample[UpscanSuccess]
           val updatedUpscanSuccess =
-            upscanSuccess.copy(uploadDetails = Map("fileName" -> supportingEvidence.fileName))
+            upscanSuccess.copy(uploadDetails = Map("fileName" -> "new-file.pdf"))
```
Comment (mahesh-hmrc) [inline test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala:1251]: We seem to be missing tests completely for the duplicate file branch [here](https://github.com/hmrc/cgt-property-disposals-frontend/pull/1364/changes#diff-7b4979e9ea8c57ea4af90d570a1af45a5b83df63bd6c1c3be09502eada9d00a7R351)

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/supportingevidence/upload.scala.html (hunk 2 above)
Comment (ZaraFarzan) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/supportingevidence/upload.scala.html]: Done

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala (hunk 3 above)
Comment (ZaraFarzan) [reply test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala:1251]: Done

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceController.scala (hunk 1 above)
Comment (ZaraFarzan) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceController.scala]: Done

**Review by mahesh-hmrc — APPROVED** [review]
